modules/cnvkit/1.0/cnvkit.smk

```diff
@@ -557,26 +558,191 @@ rule _cnvkit_to_seg:
+
+    call_cns = str(rules._run_cnvkit_call_vcf.output.cns).replace("{capture_space}", this_space[0])
+    scatter = str(rules._run_cnvkit_scatter.output.png).replace("{capture_space}", this_space[0])
+    diagram = str(rules._run_cnvkit_diagram.output.pdf).replace("{capture_space}", this_space[0])
+    breaks = str(rules._cnvkit_breaks.output.breaks).replace("{capture_space}", this_space[0])
+    gene_seg = str(rules._cnvkit_genemetrics_seg.output.genemetrics).replace("{capture_space}", this_space[0])
+    geneList = str(rules._cnvkit_trusted_genes_cna.output.trusted_genes).replace("{capture_space}", this_space[0])
+    sex = str(rules._cnvkit_infer_sex.output.sex).replace("{capture_space}", this_space[0])
+    seg = str(rules._cnvkit_to_seg.output.seg).replace("{capture_space}", this_space[0])
+
+    return{
+        "call_cns": call_cns,
+        "scatter": scatter,
+        "diagram": diagram,
+        "breaks": breaks,
+        "gene_seg": gene_seg,
+        "geneList": geneList,
+        "sex": sex,
+        "seg": seg
+    }
+
 # Symlinks the final output files into the module results directory (under '99-outputs/')
 rule _cnvkit_output:
     input:
-        call_cns = str(rules._run_cnvkit_call_vcf.output.cns),
-        scatter = str(rules._run_cnvkit_scatter.output.png),
-        diagram = str(rules._run_cnvkit_diagram.output.pdf),
-        breaks = str(rules._cnvkit_breaks.output.breaks),
-        gene_seg = str(rules._cnvkit_genemetrics_seg.output.genemetrics),
-        geneList = str(rules._cnvkit_trusted_genes_cna.output.trusted_genes),
-        sex = str(rules._cnvkit_infer_sex.output.sex),
-        seg = str(rules._cnvkit_to_seg.output.seg)
+        unpack(_cnvkit_drop_capture_space_wc)
     output:
-        call_cns = CFG["dirs"]["outputs"] + "wildcard_capture/BAF_cns/{seq_type}--{genome_build}/{capture_space}/{tumour_id}.call.cns",
-        scatter = CFG["dirs"]["outputs"] + "wildcard_capture/scatter/{seq_type}--{genome_build}/{capture_space}/{tumour_id}_scatter.png",
-        diagram = CFG["dirs"]["outputs"] + "wildcard_capture/diagram/{seq_type}--{genome_build}/{capture_space}/{tumour_id}_diagram.pdf",
-        breaks = CFG["dirs"]["outputs"] + "wildcard_capture/breaks/{seq_type}--{genome_build}/{capture_space}/{tumour_id}_genebreaks.txt",
-        gene_seg = CFG["dirs"]["outputs"] + "wildcard_capture/geneMetrics/{seq_type}--{genome_build}/{capture_space}/{tumour_id}_geneSeg.txt",
-        geneList = CFG["dirs"]["outputs"] + "wildcard_capture/geneList/{seq_type}--{genome_build}/{capture_space}/{tumour_id}_genebreaks.txt",
-        sex = CFG["dirs"]["outputs"] + "wildcard_capture/sex/{seq_type}--{genome_build}/{capture_space}/{tumour_id}_genebreaks.txt",
-        seg = CFG["dirs"]["outputs"] + "wildcard_capture/seg/{seq_type}--{genome_build}/{capture_space}/{tumour_id}.seg"
+        call_cns = CFG["dirs"]["outputs"] + CFG["output"]["cns"]["call"],
```

Comment:
What's the rationale for moving the output file name specification to the config? This isn't something we do in any other modules I can think of.

Reply:
Right, we don't do it in other modules yet - but I thought it might be a good starting point to start switching. This gives any downstream tool, like GAMBLR for example, or a merging snakefile, a way to know upfront the path to the output files, and the file extension of every output. This is easy to pull back into snakefile, but I wanted to see what others think whether this can simplify developing the downstream workflows/functions

Reply:
Does Python code work when embedded in the config like that?

Reply:
Yes :heavy_check_mark:  I have run each of these modified modules on the demo sample and can confirm it works and executes as expected, generating files at the expected locations and does not give python/snakemake errors.

Reply:
It won't work in R though. Can we change it to store the full path instead of the path and Python code or is that not possible?

Reply:
I think it should be possible! I'll modify and test it out. Will push with the PR comments. Thanks!